doc/ir-semantics.text

```diff
@@ -61,12 +61,13 @@ v :=     values
 osr-map ::=    osr mapping declaration
 | const x = e
 | mut x
-| mut x = x
+| mut x = &x
+| mut x = e
 
 # Natural operational semantics
 
 E ::= (x ↦ val v | x ↦ ref a | x ↦ ⊥)*  lexical environment
-H ::= (l ↦ v)                           mutable heap
+H ::= (a ↦ v)                           mutable heap
```

Comment:
While I'm at it, nitpick at how "mutable heap" doesn't line up with "lexical environment."